src/appleseed/renderer/modeling/bsdf/microfacethelper.h

```diff
@@ -166,26 +174,36 @@ class MicrofacetBRDFHelper
         const foundation::Vector3f&     outgoing,
         const foundation::Vector3f&     incoming,
         FresnelFun                      f,
-        const float                     cos_in,
-        const float                     cos_on,
         Spectrum&                       value)
     {
-        const foundation::Vector3f h = foundation::normalize(incoming + outgoing);
-        const foundation::Vector3f m = shading_basis.transform_to_local(h);
-        const foundation::Vector3f wo = shading_basis.transform_to_local(outgoing);
+        foundation::Vector3f wo = shading_basis.transform_to_local(outgoing);
+        foundation::Vector3f wi = shading_basis.transform_to_local(incoming);
+
+        if (Flip)
```

Comment:
I'm still slightly unsure about what `Flip` means. Why use `std::abs()` instead of negating Y components?

Reply:
I moved the flipping code from the glossy brdf here. The names come from there.
What this does is place wi and wo in the same hemisphere as the shading normal.

Reply:
Ok. Maybe you can add this in a comment?